protocol-specs/generators/lisk_codec/index.js

```diff
@@ -0,0 +1,555 @@
+				type: 'object',
+				properties: {
+					state: {
+						dataType: 'boolean',
+						fieldNumber: 1,
+					},
+				},
+			},
+		},
+		booleanFalse: {
+			object: {
+				state: false,
+			},
+			schema: {
+				type: 'object',
+				properties: {
+					state: {
+						dataType: 'boolean',
+						fieldNumber: 1,
+					},
+				},
+			},
+		},
+	};
+
+	const booleanTrueEncoded = Boolean.encode(input.booleanTrue.object).finish();
+	const booleanFalseEncoded = Boolean.encode(input.booleanFalse.object).finish();
+
+	return {
+		description: 'Encoding of boolean types',
+		config: {
+			network: 'devnet',
+		},
+		input: {
+			booleanTrue: input.booleanTrue,
+			booleanFalse: input.booleanFalse,
+		},
+		output: {
+			booleanTrue: booleanTrueEncoded.toString('hex'),
+			booleanFalse: booleanFalseEncoded.toString('hex'),
+		},
+	};
+};
+
+const generateValidStringEncodings = () => {
+	const input = {
+		string: {
+			object: {
+				data: 'Checkout Lisk SDK!',
+			},
+			schema: {
+				type: 'object',
+				properties: {
+					data: {
+						dataType: 'string',
+						fieldNumber: 1,
+					},
+				},
+			},
+		},
+		emptyString: {
+			object: {
+				data: '',
+			},
+			schema: {
+				type: 'object',
+				properties: {
+					data: {
+						dataType: 'string',
+						fieldNumber: 1,
+					},
+				},
+			},
+		},
+	};
+
+	const stringEncoded = String.encode(input.string.object).finish();
+	const emptyStringEncoded = String.encode(input.emptyString.object).finish();
+
+	return {
+		description: 'Encoding of string types',
+		config: {
+			network: 'devnet',
+		},
+		input: {
+			string: input.string,
+			emptyString: input.emptyString,
+		},
+		output: {
+			string: stringEncoded.toString('hex'),
+			emptyString: emptyStringEncoded.toString('hex'),
+		},
+	};
+};
+
+
+const generateValidBytesEncodings = () => {
+	const input = {
+		bytes: {
+			object: {
+				address: Buffer.from('e11a11364738225813f86ea85214400e5db08d6e', 'hex').toString('hex'),
+			},
+			schema: {
+				type: 'object',
+				properties: {
+					address: {
+						dataType: 'bytes',
+						fieldNumber: 1,
+					},
+				},
+			},
+		},
+		emptyBytes: {
+			object: {
+				address: Buffer.from('').toString('hex'),
+			},
+			schema: {
+				type: 'object',
+				properties: {
+					data: {
+						dataType: 'string',
+						fieldNumber: 1,
+					},
+				},
+			},
+		},
+	};
+
+	const bytesEncoded = Bytes.encode(input.bytes.object).finish();
+	const emptyBytesEncoded = Bytes.encode(input.emptyBytes.object).finish();
+
+	return {
+		description: 'Encoding of bytes types',
+		config: {
+			network: 'devnet',
+		},
+		input: {
+			bytes: input.bytes,
+			emptyBytes: input.emptyBytes,
+		},
+		output: {
+			bytes: bytesEncoded.toString('hex'),
+			emptyBytes: emptyBytesEncoded.toString('hex'),
+		},
+	};
+};
+
+const generateValidObjectEncodings = () => {
+	const input = {
+		object: {
+			object: {
+				address: Buffer.from('e11a11364738225813f86ea85214400e5db08d6e', 'hex').toString('hex'),
```

Comment:
```suggestion
				address: 'e11a11364738225813f86ea85214400e5db08d6e',
```

Reply:
This was for simplifying the output as the default JSON.stringified version was not favoured by some reviewers. It has been corrected already.